src/main/scala/com/microsoft/hyperspace/actions/RefreshDeleteActionBase.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.actions
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+
+import com.microsoft.hyperspace.HyperspaceException
+import com.microsoft.hyperspace.index.{Content, IndexDataManager, IndexLogManager}
+
+private[actions] abstract class RefreshDeleteActionBase(
+    spark: SparkSession,
+    logManager: IndexLogManager,
+    dataManager: IndexDataManager)
+    extends RefreshActionBase(spark, logManager, dataManager) {
+
+  /**
+   * Validate index has lineage column and it is in active state for refreshing and
+   * there are some deleted source data file(s).
+   */
+  override def validate(): Unit = {
+    super.validate()
+    if (!previousIndexLogEntry.hasLineageColumn(spark)) {
+      throw HyperspaceException(
+        "Index refresh (to handle deleted source data) is " +
+          "only supported on an index with lineage.")
+    }
+
+    if (deletedFiles.isEmpty || deletedFiles.toSet.equals(
+          previousIndexLogEntry.source.plan.properties.
+            relations.head.data.properties.excluded.toSet)) {
+      throw HyperspaceException("Refresh aborted as no deleted source data file found.")
```

Comment:
**To Reviewers:** This part of validate is changed (it now compares deleted files to the list of already excluded files).

Reply:
What if a user want "optimize" refresh after "quick"(OnRead) refresh and no data change?

Reply:
The scenario you pointed to should be addressed by a call to `optimize`, not `refresh`.
`optimize` and `refresh` are two different APIs and they should have different `validate` checks (at least based on current design). The `validate` function here is used with `refresh` when dealing with deleted files and that is why at least one new deleted file should be found comparing to previous case.

Reply:
Sorry for the lack of clarity. I meant RefreshDeleteAction after DeleteOnReadAction without data change.
In this case, `validate` of `RefreshDeleteAction ` will be aborted because of `deletedFiles.toSet.equals(previousIndexLogEntry.excludedFiles.toSet)`?

Reply:
I see, that is a fair scenario. Thanks for bringing it up.
I changed the code by:
1. Moved `excluded` files check of `validate` to `DeleteOnRead` class.
2. Added a new test case under `RefreshIndexTests` to validate this scenario.